test-data/unit/check-literal.test

```diff
@@ -197,10 +211,26 @@ from typing_extensions import Literal
 a: Literal[""]
 b: Literal["  foo bar  "]
 c: Literal['  foo bar  ']
+d: Literal["foo"]
+e: Literal['foo']
 
 reveal_type(a)  # E: Revealed type is 'Literal['']'
 reveal_type(b)  # E: Revealed type is 'Literal['  foo bar  ']'
 reveal_type(c)  # E: Revealed type is 'Literal['  foo bar  ']'
+reveal_type(d)  # E: Revealed type is 'Literal['foo']'
+reveal_type(e)  # E: Revealed type is 'Literal['foo']'
+
+def f1(x: Literal[""]) -> Literal[""]: pass
+def f2(x: Literal["  foo bar  "]) -> Literal["  foo bar  "]: pass
+def f3(x: Literal['  foo bar  ']) -> Literal['  foo bar  ']: pass
+def f4(x: Literal["foo"]) -> Literal["foo"]: pass
+def f5(x: Literal['foo']) -> Literal['foo']: pass
+
+reveal_type(f1)  # E: Revealed type is 'def (x: Literal['']) -> Literal['']'
+reveal_type(f2)  # E: Revealed type is 'def (x: Literal['  foo bar  ']) -> Literal['  foo bar  ']'
+reveal_type(f3)  # E: Revealed type is 'def (x: Literal['  foo bar  ']) -> Literal['  foo bar  ']'
+reveal_type(f4)  # E: Revealed type is 'def (x: Literal['foo']) -> Literal['foo']'
+reveal_type(f5)  # E: Revealed type is 'def (x: Literal['foo']) -> Literal['foo']'
```

Comment:
Test calls to a function accepting a union of string literal types. Call it with each supported literal and also union argument types.

Reply:
Added to a test named 'testLiteralCallingUnionFunction'.